Feature/VOST #13 native splash screen

OldMetalmind opened the pull request:
Solves https://github.com/vostpt/mobile-app/issues/13

--

Note:
While this Pull Request adds the functionality of Native Splash screen of iOS and Android, it will be necessary to fit it with the [Flare Splash Screen](https://github.com/vostpt/mobile-app/issues/14) after the two PRs are done.

OldMetalmind committed:
Add first iteration of a Native splash screen

OldMetalmind committed:
Add first iteration of a Native splash screen

OldMetalmind committed:
Add first iteration of a Native splash screen

OldMetalmind committed:
Fix conflig in Podfile.lock, accept incomming cocoapods version

OldMetalmind committed:
VOST-14 Adds Flare dependency and first animation

OldMetalmind committed:
Add VSCode snippets

Adds two snippets _page and _pagebloc to be used when creating
interface for Pages and respective Bloc

OldMetalmind committed:
Add VSCode snippets

Adds two snippets _page and _pagebloc to be used when creating
interface for Pages and respective Bloc

OldMetalmind committed:
Merge branch 'feature/VOST-14_Flare_Splash_Screen' of github.com:OldMetalmind/mobile-app into feature/VOST-14_Flare_Splash_Screen

OldMetalmind committed:
Adds new navigation to SplashPage

Now the "/" navigation is to open SplashPage

OldMetalmind committed:
Create SplashPage file and class

This will create a page with a blue background and a flare animation

OldMetalmind committed:
Update VSCode snippet

OldMetalmind committed:
Adds native splash

This will add the Android and iOS native splash screen to open with
the white VOST logo with a blue background, before any other screen
is opened

OldMetalmind committed:
Adds native splash

This will add the Android and iOS native splash screen to open with
the white VOST logo with a blue background, before any other screen
is opened

OldMetalmind committed:
Resize Android vost logo used in Splash Screen, update dependencies

Also, setting the minimum android version to 23 so we can resize the
images in the Splash screen and use other new features of Android.

OldMetalmind committed:
Resize Android vost logo used in Splash Screen, update dependencies

Also, setting the minimum android version to 23 so we can resize the
images in the Splash screen and use other new features of Android.

OldMetalmind committed:
Adds Splash navigation to Home after a timer.

After a couple a seconds the Splash screen will navigate to the Home

miguelpruivo commented on ios/Podfile.lock:
```diff
@@ -19,4 +19,4 @@ SPEC CHECKSUMS:
 PODFILE CHECKSUM: ebd43b443038e611b86ede96e613bd6033c49497
 
-COCOAPODS: 1.7.1
+COCOAPODS: 1.6.0
```
Algum motivo em particular para estarmos a usar 1.6.0?

OldMetalmind replied:
No reason, I didn't notice it downgraded it. 
I'll update the PR with the change

OldMetalmind replied:
Let's fix it with this https://github.com/vostpt/mobile-app/issues/59

miguelpruivo commented on ios/Runner/AppDelegate.swift:
```diff
@@ -7,6 +7,7 @@ import Flutter
     _ application: UIApplication,
     didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?
   ) -> Bool {
+    UIApplication.shared.isStatusBarHidden = false;
```
Acho que isto é o comportamento por default.

OldMetalmind replied:
Since I followed some guides, all of them told me to add this line.
I'll test with and without and update this PR

miguelpruivo commented on ios/Runner/Info.plist:
```diff
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>UIStatusBarHidden</key>
```
Em princípio se apagares isto, a linha de código que mete a status bar novamente visível deixará de ser precisa (caso se pretenda que esteja visível, que é o comportamento por omissão).

OldMetalmind replied:
I'll test it out, thank you.

miguelpruivo commented on pubspec.yaml:
```diff
@@ -43,4 +43,7 @@ dev_dependencies:
+  uses-material-design: true
+
+  assets:
+    - assets/images/vost_logo_white.png
```
Se este logo não for usado dentro da app e for só no splash nativo, não é necessário estar aqui. Mas acredito que seja usado, é só para confirmar.

OldMetalmind replied:
I left it out since it really might be used inside the app, also, this way, it is the source of truth for the Android and iOS native splash.
Can we leave it?

miguelpruivo replied:
Sure! That's what I thought.

miguelpruivo reviewed (commented):
@OldMetalmind vê os comentários. Aproveito também para alertar que se deve rectificar alguns files que deveriam estar no _.gitignore_.

Commit:
Merge pull request #58 from OldMetalmind/feature/VOST-14_Flare_Splash_Screen

Feature/VOST #14 Flare Splash Screen

OldMetalmind committed:
Removes unecessary default values

OldMetalmind committed:
Merge branch 'feature/VOST-13_Native_Splash_Screen' of github.com:OldMetalmind/mobile-app into feature/VOST-13_Native_Splash_Screen

OldMetalmind committed:
Remove declaration of already default values

Variables where set that are unnecessary to set since they don't change
anything

OldMetalmind committed:
Update .gitignore file to exclude Eclipse IDE files
